File: Data Structures & Algorithms/binary-search/submission-3.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        while (l <= r) {
            int idx = (l + r) / 2;
            if (nums[idx] > target) {
                r = idx - 1;
            } else if (nums[idx] < target) {
                l = idx + 1;
            } else {
                return idx;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        starts = set()
        numsSet = set(nums)
        for n in nums:
            if n - 1 not in numsSet:
                starts.add(n)
        res = 0
        for s in starts:
            l = 1
            curr = s
            while curr + 1 in nums:
                l += 1
                curr += 1
            res = max(res, l)
        return res